umodbus/async/serial.py

```diff
@@ -0,0 +1,304 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2019, Pycom Limited.
+#
+# This software is licensed under the GNU GPL version 3 or any
+# later version, with permitted additional terms. For more information
+# see the Pycom Licence v1.0 document supplied with this file, or
+# available at https://www.pycom.io/opensource/licensing
+#
+
+# system packages
+from ..sys_imports import struct
+from ..sys_imports import List, Literal, Optional, Tuple, Union
+
+try:
+    import uasyncio as asyncio
+    from machine import UART, Pin
+except ImportError:
+    import asyncio
+
+# custom packages
+from .. import functions, const as Const
+from .common import AsyncRequest
+from .modbus import Modbus
+
+
+class SerialServer(AbstractServerInterface):
+    def __init__(self,
+                 uart_id:int = 1,
+                 baudrate:int = 9600,
+                 data_bits:int = 8,
+                 stop_bits:int = 1,
+                 parity=None,
+                 pins: Tuple[int, int] = (13, 14),
```

Comment:
please keep it as 
```suggestion
                 pins: Tuple[int, int] = None,
```
to explicitly throw an error if no pins are specified